app/assets/javascripts/discourse/app/services/sidebar-state.js

```diff
@@ -0,0 +1,74 @@
+import Service from "@ember/service";
+import { tracked } from "@glimmer/tracking";
+import I18n from "I18n";
+
+import BaseCustomSidebarSection from "discourse/lib/sidebar/base-custom-sidebar-section";
+import BaseCustomSidebarPanel from "discourse/lib/sidebar/base-custom-sidebar-panel";
+import BaseCustomSidebarSectionLink from "discourse/lib/sidebar/base-custom-sidebar-section-link";
+
+class MainSidebarPanel {
+  sections = [];
+
+  get key() {
+    return "main";
+  }
+
+  get switchButtonLabel() {
+    return I18n.t("sidebar.panels.forum.label");
+  }
+
+  get switchButtonIcon() {
+    return "random";
+  }
+
+  get switchButtonDefaultUrl() {
+    return "/";
+  }
+}
+
+const COMBINED_MODE = "combined";
+const SEPARATED_MODE = "separated";
+
+export default class SidebarState extends Service {
+  @tracked currentPanelKey = "main";
+  @tracked panels = [new MainSidebarPanel()];
+  @tracked mode = COMBINED_MODE;
+
+  addPanel(func) {
+    const panelClass = func.call(this, BaseCustomSidebarPanel);
+    this.panels.push(new panelClass());
+  }
+
+  setPanel(name) {
+    this.currentPanelKey = name;
+  }
+
+  get currentPanel() {
+    return this.panels.find((panel) => panel.key === this.currentPanelKey);
+  }
+
+  setSeparatedMode() {
+    this.mode = SEPARATED_MODE;
+  }
+
+  setCombinedMode() {
+    this.mode = COMBINED_MODE;
+  }
+
+  get combinedMode() {
+    return this.mode === COMBINED_MODE;
+  }
+
+  addSidebarSection(func, panelKey) {
```

Comment:
I think we need to think about what is the purpose of this service object and what state is it responsible for. It feels very odd to me that the method to register a sidebar section is part of the `SidebarState` service object. To me, the `SidebarState` service object sounds like it should only allow us to set and retrieve the state of the sidebar.